auth.go

```diff
@@ -51,6 +51,7 @@ func NewError(base error, detail string) error {
 // A Permit is an opaque binary string that encodes domain-specific roles.
 // This could be a string of bits with special meanings, or a blob of JSON, etc.
 type Permit struct {
+	name string
```

Comment:
What is Permit.name? It would need a very good reason to justify it's existence.